README/README.CXXMODULES.md

```diff
@@ -0,0 +1,486 @@
+```cpp
+// ROOT prompt
+root [] namespace foo { };struct S;
+root [] foo::bar/*store parsing state*/
+        gSystem->Load("Foo");
+        // More scaffolding.
+        extern int __Cling_Autoloading_Map;
+        namespace foo{struct __attribute__((annotate("$clingAutoload$Foo.h"))) bar;}
+        struct __attribute__((annotate("$clingAutoload$Foo.h"))) S;
+        // More initialization scaffolding.
+        /*restore parsing state*/ baz1 /*store parsing state*/
+        #include <Foo.h>/*restore parsing state*/;
+```
+
+To recap, unfortunately, ROOT PCH is not extendable; ROOTMAP require a lot of
+maintenance and goes on a very untested codepath, while RDICT has a very limited
+scope. The three features require a lot of mechanisms to work together and the
+corner cases are very many. The interaction between some of the features are
+not well defined.
+
+
+## From C++ Modules to Dictionaries
+
+C++ Modules have native capabilities to avoid reparsing. It combines all
+home-grown solutions to avoid the costly operation at industry quality.
+
+Currently, when ROOT is built with `-Druntime_cxxmodules=On` it gives priority to
+C++ Module files (real *pcm* files). If such a file is present it reads all
+necessary information from it. If no such file is present ROOT falls back to the
+standard information flow.
+
+### Observable differences from 'standard' ROOT
+
+As always, ROOT is (mostly) API and ABI compatible. C++ Modules-aware ROOT is no
+different. There are several differences which can be noticed:
+  * \*modulemap files in $ROOTSYS/include -- those files are used by rootcling to
+  put a set of header files in a single pcm file. For example, all related
+  headers of *libGeom* are persisted in *Geom.pcm*. There are a few notable
+  examples, which are specific to the way we build ROOT. In certain cases we
+  want some header files to be compiled within C context or with RTTI on/off.
+  That's mostly for bootstrapping ROOT (aka rootcling stage1).
+  * modulemap.overlay.yaml -- automatically created virtual filesystem overlay
+  file. This file contains introduces C++ Modules for third-party dependencies.
+  For example, to 'modularize' glibc for ROOT we need to place a modulemap
+  file in (usually) `/usr/include`. This folder is not writable on many
+  platforms. The vfs file tells the compiler to pretend there is a file at a
+  specific location. This way we 'mount' `/usr/include/module.modulemap`
+  non-invasively. The reasons why we need to extend the C++ modules support
+  beyond ROOT is described bellow.
+  * rootcling creates a new binary artifact *Name.pcm* after the library name --
+  this is a temporary solution for the current technology preview. Once we
+  advance further the implementation we will only create Name.pcm without the
+  other 2 artifacts.
+  * Preloads all \*pcm files at start up time -- this currently is the only
+  remaining bottleneck which introduces a relatively small performance overhead
+  described bellow.
```

Comment:
Clarify that this is a bottleneck for ROOT's startup time! And will be dwarfed by e.g. frameworks' startup and header parsing time.

Reply:
Done.